library/strings/wildcard_pattern_matching.hpp

```diff
@@ -0,0 +1,49 @@
+/** @file */
+#pragma once
+template <class T>
+vector<T> make_powers(const T& v) {
+  int n = sz(v);
+  vector pws(3, T(n));
+  rep(i, 0, n) {
+    pws[0][i] = v[i];
+    pws[1][i] = v[i] * v[i];
+    pws[2][i] = v[i] * v[i] * v[i];
+  }
+  return pws;
+}
+/**
+ * @brief wildcard pattern matching
+ * @note the cumulative values for comparision will be on the order of
+ * (max element)^4
+ *
+ * @param haystack is the string|array|vector to search in where the value `0`
```

Comment:
ah yes, good interface; I was going to suggest this. Basically, make the user handle the wildcard